### PR TITLE
fix(masthead): lazy loading megamenu and side nav

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#9582

### Description
Currently the Masthead loads both the SideNav and Megamenu components at the same time, even if they haven't been opened yet. This causes the amount of nodes to increase considerably right away. In order to mitigate this, a fast solution is to dynamically load the required components *only* when the user has interacted with the menus.

This means that the mobile version wouldn't be loading the Megamenu, and the desktop wouldn't load SideNav until/if required. 

Another optimization that was done ensures that the Megamenu component wouldn't get loaded into the DOM unless a user has clicked on the menu beforehand. Then and only then, the megamenu would be loaded in. Once the menu is closed, the component gets removed from the DOM to ensure the total node count doesn't shoot up.

Using @juanleon9's NodeCount script on the `dds-masthead-container` (which contains both `dds-side-nav` and `dds-megamenu`) these are the following stats:

* Megamenu and SideNav not loaded:
 `NODES 1319 COMMENTS 316 INTERS 671`

* Megamenu loaded in and opened / SideNav not loaded (essentially Desktop view):
  `NODES 2445 COMMENTS 553 INTERS 1224`

The megamenu would go be removed from the DOM right after it closes, so the Masthead will always remain at around `1319`

Seems like the total nodes each menu brings in is about:
* Megamenu: `2744`
* SideNav: `4021`

At least with this PR we're saving about `6765` nodes from populating the page, at least right away. 

### Changelog

**Changed**

- lazy loading `side-nav` and `megamenu` components only when user interacts with them

### CODE REVIEW

File: packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts [hunk 1]
```diff
@@ -134,6 +134,14 @@ class DDSMegaMenuTopNavMenu extends DDSTopNavMenu {
         forEach(doc.querySelectorAll((this.constructor as typeof DDSMegaMenuTopNavMenu).selectorOverlay), item => {
           (item as DDSMegaMenuOverlay).active = this.expanded;
         });
+
+        import(`./megamenu-left-navigation`);
+        import('./megamenu-category-link');
+        import('./megamenu-category-group');
+        import('./megamenu-category-group-copy');
+        import('./megamenu-link-with-icon');
+        import('./megamenu-overlay');
```
Comment (andy-blum) [inline packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts:143]: We'll probably need to add a way to track once these have been imported, so we don't try to re-import everytime we update the 'expanded' property on a top-nav-menu item

**Comment by jeffchew** [comment]
perhaps there's a babel config for this? 
image

**Comment by andy-blum** [comment]
https://babeljs.io/docs/en/babel-plugin-syntax-dynamic-import

> NOTE: This plugin is included in @babel/preset-env, in [ES2020](https://github.com/tc39/proposals/blob/master/finished-proposals.md)

File: packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts (hunk 1 above)
Comment (IgnacioBecerra) [reply packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts]: Good idea, I added a boolean check before importing the components.

**Comment by IgnacioBecerra** [comment]
After discussing with @andy-blum, we agreed an even greater optimization would be to render only one megamenu at a time, and remove it from the DOM once it is closed. 

With this new method, we are now down to `NODES 1319 COMMENTS 316 INTERS 671` whenever a megamenu is closed, and only `NODES 2445 COMMENTS 553 INTERS 1224` if a megamenu is open.

**Comment by andy-blum** [comment]
@IgnacioBecerra Can you rebase this onto `feat/masthead-v2` to pull in the v2 story & other updates?

**Comment by andy-blum** [comment]
I had too many problems rebasing this, so I manually duplicated the changes to a new PR, #9602

**Comment by IgnacioBecerra** [comment]
Closing this PR in favor of Andy's remade PR.

**Pull request closed without merging**